### PR TITLE
Add components for setting redshift & viewing redshift history on source page

### PR DESCRIPTION
**Pull request by acrellin**
Closes https://github.com/skyportal/skyportal/issues/1012

redshift_history

### CODE REVIEW

**Comment by acrellin** [comment]
@kmshin1397 @dannygoldstein  Travis is green, sole GA failure is unrelated: `skyportal/tests/frontend/test_source_list.py::test_source_filtering_and_pagination`

File: skyportal/handlers/api/candidate.py [hunk 1]
```diff
@@ -1,4 +1,6 @@
 import arrow
+from copy import copy
+import datetime
```
Comment (profjsb) [inline skyportal/handlers/api/candidate.py:3]: As per PEP8, import of python builtins should come before 3rd party imports (arrow)

Comment (acrellin) [reply skyportal/handlers/api/candidate.py]: Done

**Review by dannygoldstein — CHANGES_REQUESTED** [review]
Redshifts should have permissions moderated at the group level like classifications. To facilitate this, we may want to have a separate Redshift model, and a RedshiftGroup model.

**Comment by acrellin** [comment]
@dannygoldstein in the future please make such requirements explicit in the
relevant issue/story

**Comment by acrellin** [comment]
> Redshifts should have permissions moderated at the group level like classifications. To facilitate this, we may want to have a separate Redshift model, and a RedshiftGroup model.

@dannygoldstein why was this not brought up when we implemented setting redshift via comments?

**Comment by acrellin** [comment]
> > Redshifts should have permissions moderated at the group level like classifications. To facilitate this, we may want to have a separate Redshift model, and a RedshiftGroup model.
> 
> @dannygoldstein why was this not brought up when we implemented setting redshift via comments? I propose that we defer that to a future PR

@dannygoldstein please file a detailed issue so that this can be implemented later

**Review by kmshin1397 — APPROVED** [review]
This all seems good to me. Agreed with @acrellin that data model changes to facilitate permissions can be a separate PR. Seems to me like just getting this PR in would be a big enhancement over what we have currently even without those permission changes.

**Comment by acrellin** [comment]
okay @kmshin1397 @dannygoldstein this should be good to go

**Comment by acrellin** [comment]
Sole Travis failure is the unrelated, known flaky skyportal/tests/frontend/test_recent_sources.py::test_recently_saved_candidate

**Review by kmshin1397 — APPROVED** [review]

**Pull request merged**